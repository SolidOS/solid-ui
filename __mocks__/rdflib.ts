export const Util =  {
  mediaTypeClass: sym
}

export function sym () {
  return {
    dir: sym,
    uri: 'uri',
    value: '',
    elements: [],
    doc: () => {
      return {
        equals: () => {},
        value: ''
      }
    },
    sameTerm: () => false
  }  
}  

export const uri = {
  join: () => {}
}
export function st() {
}

export function literal() {
}
export function graph() {
  return {
    any: () => {
      return sym()
    },
    each: () => {
      return []
    },
    match: () => {
      return []
    },
    query: () => {
      return []
    },
    fetcher: {
      load: () => {},
      nowOrWhenFetched: () => Promise.resolve()
    },
    findTypeURIs: () => {
      return []
    },
    findSuperClassesNT: () => {
      return []
    },
    bottomTypeURIs: () => {
      return []
    },
    statementsMatching: () => {
      return []
    },
    sym
  }
}
export function fetcher() {
  return {}
}
export class Fetcher {
  requested: any
  constructor() {
    this.requested = {}
  }
  load () {
  }
}
export function namedNode() {
  return {}
}
export function NamedNode() {
}
export function Namespace() {
  return () => {}
}
export class UpdateManager {
  editable() {}
  update() {}
}
