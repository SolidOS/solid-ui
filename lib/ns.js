"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
var _typeof = require("@babel/runtime/helpers/typeof");
var _solidNamespace = _interopRequireDefault(require("solid-namespace"));
var $rdf = _interopRequireWildcard(require("rdflib"));
function _interopRequireWildcard(e, t) { if ("function" == typeof WeakMap) var r = new WeakMap(), n = new WeakMap(); return (_interopRequireWildcard = function _interopRequireWildcard(e, t) { if (!t && e && e.__esModule) return e; var o, i, f = { __proto__: null, "default": e }; if (null === e || "object" != _typeof(e) && "function" != typeof e) return f; if (o = t ? n : r) { if (o.has(e)) return o.get(e); o.set(e, f); } for (var _t in e) "default" !== _t && {}.hasOwnProperty.call(e, _t) && ((i = (o = Object.defineProperty) && Object.getOwnPropertyDescriptor(e, _t)) && (i.get || i.set) ? o(f, _t, i) : f[_t] = e[_t]); return f; })(e, t); }
// Namespaces we commonly use and have common prefixes for around Solid
//

// Delegate to this which takes RDFlib as param.

module.exports = (0, _solidNamespace["default"])($rdf);
// export * from ns2

// ends
//# sourceMappingURL=ns.js.map