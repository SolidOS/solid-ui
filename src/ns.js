// thisModule = {}

var thisModule = {}

module.exports = thisModule

// tabulator.trace = thisModule // @@@@@@@@@@

var $rdf = require('rdflib')

// This used to be a faw function (no ".expand" but that caused the module exports all to be dropped)
thisModule.expand = function (prefixed) {
  var pair = prefixed.split(':')
  if (pair.length === 0) throw new Error('Prefixed name has no colon: ' + prefixed)
  if (!(pair[0] in thisModule)) throw new Error('Unregistered namespace prefix in: ' + prefixed)
  return thisModule[pair[0]](pair[1])
}

thisModule.auth = $rdf.Namespace('http://www.w3.org/ns/auth/acl#') // @@ obsolete - use acl:
thisModule.acl = $rdf.Namespace('http://www.w3.org/ns/auth/acl#')
thisModule.arg = $rdf.Namespace('http://www.w3.org/ns/pim/arg#')
thisModule.cal = $rdf.Namespace('http://www.w3.org/2002/12/cal/ical#')
thisModule.contact = $rdf.Namespace('http://www.w3.org/2000/10/swap/pim/contact#')
thisModule.dc = $rdf.Namespace('http://purl.org/dc/elements/1.1/')
thisModule.dct = $rdf.Namespace('http://purl.org/dc/terms/')
thisModule.doap = $rdf.Namespace('http://usefulinc.com/ns/doap#')
thisModule.foaf = $rdf.Namespace('http://xmlns.com/foaf/0.1/')
thisModule.http = $rdf.Namespace('http://www.w3.org/2007/ont/http#')
thisModule.httph = $rdf.Namespace('http://www.w3.org/2007/ont/httph#')
thisModule.ical = $rdf.Namespace('http://www.w3.org/2002/12/cal/icaltzd#')
thisModule.ldp = $rdf.Namespace('http://www.w3.org/ns/ldp#')
thisModule.link = thisModule.tab = thisModule.tabont = $rdf.Namespace('http://www.w3.org/2007/ont/link#')
thisModule.log = $rdf.Namespace('http://www.w3.org/2000/10/swap/log#')
thisModule.meeting = $rdf.Namespace('http://www.w3.org/ns/pim/meeting#')
thisModule.mo = $rdf.Namespace('http://purl.org/ontology/mo/')
thisModule.owl = $rdf.Namespace('http://www.w3.org/2002/07/owl#')
thisModule.pad = $rdf.Namespace('http://www.w3.org/ns/pim/pad#')
thisModule.patch = $rdf.Namespace('http://www.w3.org/ns/pim/patch#')
thisModule.qu = $rdf.Namespace('http://www.w3.org/2000/10/swap/pim/qif#')
thisModule.trip = $rdf.Namespace('http://www.w3.org/ns/pim/trip#')
thisModule.rdf = $rdf.Namespace('http://www.w3.org/1999/02/22-rdf-syntax-ns#')
thisModule.rdfs = $rdf.Namespace('http://www.w3.org/2000/01/rdf-schema#')
thisModule.rss = $rdf.Namespace('http://purl.org/rss/1.0/')
thisModule.sched = $rdf.Namespace('http://www.w3.org/ns/pim/schedule#')
thisModule.schema = $rdf.Namespace('http:/schema.org/') // @@ beware confusion with documents no 303
thisModule.sioc = $rdf.Namespace('http://rdfs.org/sioc/ns#')
// was - thisModule.xsd = $rdf.Namespace('http://www.w3.org/TR/2004/REC-xmlschema-2-20041028/#dt-')
thisModule.solid = $rdf.Namespace('http://www.w3.org/ns/solid/terms#')
thisModule.space = $rdf.Namespace('http://www.w3.org/ns/pim/space#')
thisModule.stat = $rdf.Namespace('http://www.w3.org/ns/posix/stat#')
thisModule.ui = $rdf.Namespace('http://www.w3.org/ns/ui#')
thisModule.vcard = $rdf.Namespace('http://www.w3.org/2006/vcard/ns#')
thisModule.wf = $rdf.Namespace('http://www.w3.org/2005/01/wf/flow#')
thisModule.xsd = $rdf.Namespace('http://www.w3.org/2001/XMLSchema#')

//  ends
