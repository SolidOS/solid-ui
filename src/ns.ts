// Namespaces we commonly use and have common prefixes for around Solid
//

const solidNamespace = require('solid-namespace') // Delegate to this which takes RDFlib as param.
const $rdf = require('rdflib')

export default solidNamespace($rdf)

// ends
