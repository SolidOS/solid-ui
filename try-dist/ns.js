"use strict";

// Namespaces we commonly use and have common prefixes for around Solid
//
var solidNamespace = require('solid-namespace'); // Delegate to this which takes RDFlib as param.


var $rdf = require('rdflib');

module.exports = solidNamespace($rdf); // ends
//# sourceMappingURL=ns.js.map