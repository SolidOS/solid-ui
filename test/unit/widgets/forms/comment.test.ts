import { silenceDebugMessages } from '../../../setup'
import { namedNode } from 'rdflib'
import ns from '../../../../src/ns'
import uiStore from '../../../../src/store'

import {
  commentField
} from '../../../../src/widgets/forms/comment'
import { clearStore } from '../../helpers/clearStore'

silenceDebugMessages()
afterEach(clearStore)

describe('Comment', () => {
  it('exists', () => {
    expect(commentField).toBeInstanceOf(Object)
  })
  it('runs', () => {
    const container = document.createElement('div')
    const already = {}
    const subject = namedNode('http://example.com/#this')
    const form = namedNode('http://example.com/#form')
    const store = namedNode('http://example.com/#store')
    const callbackFunction = jest.fn() // TODO: https://github.com/solid/solid-ui/issues/263
    uiStore.add(form, ns.rdf('type'), ns.ui('Comment'), namedNode('http://example.com/'))
    uiStore.add(form, ns.ui('contents'), namedNode('http://example.com/#bla'), namedNode('http://example.com/'))
    expect(
      commentField(
        document,
        container,
        already,
        subject,
        form,
        store,
        callbackFunction
      )
    ).toMatchSnapshot()
  })
  it('deals with missing content', () => {
    const container = document.createElement('div')
    const already = {}
    const subject = namedNode('http://example.com/#this')
    const form = namedNode('http://example.com/#form')
    const store = namedNode('http://example.com/#store')
    const callbackFunction = jest.fn() // TODO: https://github.com/solid/solid-ui/issues/263
    uiStore.add(form, ns.rdf('type'), ns.ui('Comment'), namedNode('http://example.com/'))
    expect(
      commentField(
        document,
        container,
        already,
        subject,
        form,
        store,
        callbackFunction
      )
    ).toMatchSnapshot()
  })
  it('deals with missing field params in non-bottom field type', () => {
    const container = document.createElement('div')
    const already = {}
    const subject = namedNode('http://example.com/#this')
    const form = namedNode('http://example.com/#form')
    const store = namedNode('http://example.com/#store')
    const callbackFunction = jest.fn() // TODO: https://github.com/solid/solid-ui/issues/263
    uiStore.add(form, ns.rdf('type'), namedNode('http://example.com/#custom2'), namedNode('http://example.com/'))
    uiStore.add(namedNode('http://example.com/#custom2'), ns.rdfs('subClassOf'), namedNode('http://example.com/#custom1'), namedNode('http://example.com/'))
    uiStore.add(form, ns.ui('contents'), namedNode('http://example.com/#bla'), namedNode('http://example.com/'))

    expect(
      commentField(
        document,
        container,
        already,
        subject,
        form,
        store,
        callbackFunction
      )
    ).toMatchSnapshot()
  })
  it('deals with missing container', () => {
    const already = {}
    const subject = namedNode('http://example.com/#this')
    const form = namedNode('http://example.com/#form')
    const store = namedNode('http://example.com/#store')
    const callbackFunction = jest.fn() // TODO: https://github.com/solid/solid-ui/issues/263
    uiStore.add(form, ns.rdf('type'), ns.ui('Comment'), namedNode('http://example.com/'))
    uiStore.add(form, ns.ui('contents'), namedNode('http://example.com/#bla'), namedNode('http://example.com/'))
    expect(
      commentField(
        document,
        undefined,
        already,
        subject,
        form,
        store,
        callbackFunction
      )
    ).toMatchSnapshot()
  })
  it('deals with custom style', () => {
    const already = {}
    const subject = namedNode('http://example.com/#this')
    const form = namedNode('http://example.com/#form')
    const store = namedNode('http://example.com/#store')
    const callbackFunction = jest.fn() // TODO: https://github.com/solid/solid-ui/issues/263
    uiStore.add(form, ns.rdf('type'), ns.ui('Comment'), namedNode('http://example.com/'))
    uiStore.add(form, ns.ui('style'), 'custom: true;', namedNode('http://example.com/'))
    uiStore.add(form, ns.ui('contents'), namedNode('http://example.com/#bla'), namedNode('http://example.com/'))
    expect(
      commentField(
        document,
        undefined,
        already,
        subject,
        form,
        store,
        callbackFunction
      )
    ).toMatchSnapshot()
  })
})
