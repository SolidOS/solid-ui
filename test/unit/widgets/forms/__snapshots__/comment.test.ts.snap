// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment deals with custom style 1`] = `
<div>
  <p
    style="custom: true;"
  >
    &lt;http://example.com/#bla&gt;
  </p>
</div>
`;

exports[`Comment deals with missing container 1`] = `
<div>
  <p>
    &lt;http://example.com/#bla&gt;
  </p>
</div>
`;

exports[`Comment deals with missing content 1`] = `
<div>
  <p>
    Error: No contents in comment field.
  </p>
</div>
`;

exports[`Comment deals with missing field params in non-bottom field type 1`] = `
<div>
  <p>
    &lt;http://example.com/#bla&gt;
  </p>
</div>
`;

exports[`Comment runs 1`] = `
<div>
  <p>
    &lt;http://example.com/#bla&gt;
  </p>
</div>
`;
